raft-tech/TANF-app
# Devops/2337 deploy develop on pr merge

## Summary of Changes
_This tracks pushes to the `develop` branch and triggers a build and deploy to the develop environment, followed by kicking off Cypress end to end integration testing against the newly deployed env_
Pull request closes #2337 

## How to Test
_This will set up a deployment pipeline that only tracks the develop branch_
Future tests for this will be when a PR merge (Including this one) kicks off a build, deploy, and test pipeline.
To test the functionality prior to this PR, instead of tracking the `develop` branch, this `devops/2337-deploy-develop-on-pr-merge` was put in as a placeholder, but still deployed to the develop environment.

This [GitHub Action](https://github.com/raft-tech/TANF-app/actions/runs/4125507980) kicked off the [CircleCI Pipeline](https://app.circleci.com/pipelines/github/raft-tech/TANF-app/12444). Tests against the newly deployed environment can be seen as part of [that pipeline](https://app.circleci.com/pipelines/github/raft-tech/TANF-app/12444/workflows/cb76f5b1-ab78-43c7-8e1f-2569c80a104f/jobs/34374?invite=true#step-109-39)

The tracked branch was changed to `develop` before initiating this PR. The deploy and e2e tests were not [kicked off](https://app.circleci.com/pipelines/github/raft-tech/TANF-app/12446) but when this PR is merged, a corresponding GitHub Action and CircleCI build should immediately start.


## Deliverables
_This is part of a larger initiative to refine Continuous Integration and Continuous Deployment pipelines to improve developer quality of life, confidence in higher level deployments (staging and production), and enabling safe, secure, and granular updates to the live assets_

### [Deliverable 1: Accepted Features](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [x] Upon any merge to `develop` GitHub Actions kicks off a CircleCI pipeline
+ [x] That CircleCI pipeline deploys the changes to the develop environment
+ [x] After the deploy, Cypress Tests are run against the deployed environment
+ [x] The test results can be seen within the logs of the pipeline

### [Deliverable 2: Tested Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

This PR does not include frontend or backend changes and the app should function identically as before. The process should be continuously monitored by interested parties to look for improvements or breaking pipelines.

### [Deliverable 3: Properly Styled Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

This PR used existing tools and scripts and no research was done.
Future research should look at creating:
- a stateful infrastructure manifest
- artifact storage
- environment variable and secrets storage
- CI/CD pipeline simplification
